src/views/bridge-confirmation/bridge-confirmation.view.tsx

```diff
@@ -77,102 +69,55 @@ const BridgeConfirmation: FC = () => {
   const [estimatedGas, setEstimatedGas] = useState<AsyncTask<Gas, string>>({
     status: "pending",
   });
-  const estimateGasPollInterval = useRef<number>();
+  const [fadeClass, setFadeClass] = useState<string>();
   const currencySymbol = getCurrencySymbol(getCurrency());
 
-  const startGasPolling = useCallback(() => {
-    const estimateGas = () => {
-      setEstimatedGas((currentEstimatedGas) =>
-        isAsyncTaskDataAvailable(currentEstimatedGas)
-          ? { status: "reloading", data: currentEstimatedGas.data }
-          : { status: "loading" }
-      );
-      if (formData && connectedProvider.status === "successful" && tokenBalance) {
+  useEffect(() => {
+    if (connectedProvider.status === "successful" && formData && tokenBalance) {
+      const onBlock = () => {
```

Comment:
We need to do an initial call to `onBlock` (which by the way I would rename to `updateFee`), otherwise, the screen will not load until a block is emitted.

Reply:
Here is where [this change](https://github.com/0xPolygonHermez/zkevm-bridge-ui/pull/223/files#r1003513780) is relevant, to avoid the infinite loop.

Reply:
Nope, it already fires for the first time with the last mined block, so we don't need extra calls.

Reply:
This is not what I experienced

Reply:
Check this out (with audio):

https://user-images.githubusercontent.com/15896177/197716161-4bf7f399-98e5-45b0-85d3-cd747f894f3e.mp4

Reply:
It's funny because the first time it gets called it retrieves the current block, but when the listener is detached and attached again it doesn't. Check it out:

```ts
void connectedProvider.data.provider.getBlockNumber().then(console.log);
      const onBlock = (blockNumber: number) => {
        const { from, to, token, amount } = formData;
        const destinationAddress = connectedProvider.data.account;
        console.log("new block", blockNumber);
        // ...
```

You'll see that the first time `blockNumber` is the last mined block. But this happens only the first time it gets called.

Reply:
In fact, the expected behavior is to get the current block when subscribing to the `"block"` event: https://github.com/ethers-io/ethers.js/issues/1504#issuecomment-827967938

Reply:
> It's funny because the first time it gets called it retrieves the current block

Yep, I can confirm that. So the issue I found only occurs when you go back to the form and then back to the confirmation screen. Interesting buggy behavior they have there.